src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceRange.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+
+namespace Microsoft.FeatureManagement.FeatureFilters
+{
+    /// <summary>
+    /// The recurrence range specifying how long the recurrence pattern repeats
+    /// </summary>
+    public class RecurrenceRange
+    {
+        /// <summary>
+        /// The recurrence range type
+        /// </summary>
+        public string Type { get; set; } = "NoEnd";
```

Comment:
The benefit to use string here is that it can make BindParameters very simple
```
public object BindParameters(IConfiguration filterParameters)
{
    return filterParameters.Get<TimeWindowFilterSettings>() ?? new TimeWindowFilterSettings();
}
```
Currently, the recurrence evaluator will validate the filter configuration settings. This behavior is the same as the targeting evaluator. If the "Type" property becomes enum here, the validation of "Type" will be moved to "BindParameters". I think the validation process should be performed by the recurrence evaluator.

Reply:
That makes sense in terms of not separating the validation, I was also trying to see if there's a way to avoid all the individual string comparisons and if statements in the evaluator.

Reply:
This should be enum. It's my fault. I underestimated the powerful capabilities of .NET.
At first, I thought that if I used an enum, the filterParameters.Get<TimeWindowFilterSettings>() wouldn't be able to automatically convert a string in the configuration into the enum. But when I tried it out, I found out that .NET is incredible, it can automatically parse a string into an enum. So, here, we definitly should use enum.